packages/oidc_core/lib/src/managers/user_manager_base.dart

```diff
@@ -803,6 +794,8 @@ abstract class OidcUserManagerBase {
 
   @protected
   Future<void> handleTokenExpiring(OidcToken event) async {
+    settings.onTokenExpiring?.call(event);
```

Comment:
> should we still refresh using refreshToken if the user provides their own onTokenExpiring callback?
Maybe there are scenarios where the user has to do some preparatory work before refresh can happen?

Reply:
> also would it make sense to expose a stream of expiring/expired events instead of callbacks ?
Whatever you prefer. I could also add events to the existing `events` stream using new subclasses of `OidcEvent` (`OidcTokenExpiring` and `OidcTokenExpired`)

Reply:
@ahmednfwela Any news from your side? Are there any open points that prevent you from merging this PR? This is somehow important to me as I'm currently referencing my fork in our production app 🙈

Reply:
sorry I got busy the past months, I want to use a stream based approach instead of callbacks